concent_api/core/tests/test_unit_model.py

```diff
@@ -336,3 +341,91 @@ def test_that_deposit_account_is_not_removed_when_deposit_claim_is_deleted(self)
     def test_that_no_exception_is_raised_when_deposit_claim_is_valid(self):
         self.deposit_claim.clean()
         self.deposit_claim.save()
+
+
+class ProtocolVersionValidationTest(ConcentIntegrationTestCase):
+
+    def test_that_incorrect_version_of_golem_messages_in_stored_message_should_raise_validation_error(self):
+        version_of_golem_messages_in_concent = '2.13.0'
+        different_version_of_golem_messages = '1.11.1'
+
+        with override_settings(GOLEM_MESSAGES_VERSION=version_of_golem_messages_in_concent):
```

Comment:
too much setup here - you have factories, you don't have to set everything by hand.

Reply:
done

Reply:
nope, its not what we want. It is complicated. For this validation you don't have to create that many messages. Just store  two with different `GOLEM_MESSAGES_VERSION`s without creating all the clients and stuff.

Reply:
ok, clients removed

Reply:
** For this validation you don't have to create that many messages. Just store two with different GOLEM_MESSAGES_VERSIONs without creating all the clients and stuff.
**

Reply:
Did you ran this test or only saw the code? This test requires pretty big setup, because I needed to store two different golem_messages and **subtask**, and subtask requires pretty lot of setup (inc. clients, etc.) if you want to do it manually.

Now I changed it completely, it is shorter, but little bit more complicated, using `store_subtask` and patch. Hope you will like it.

BTW, there was another big problem in this test, it tested only that `ValidationError` is raised, without reason. I added it and please remember about at it in other pull requests, code should be nice, etc. but here big logic mistake was missed.

Reply:
It's ok now.